lms/djangoapps/courseware/courses.py

```diff
@@ -108,6 +108,17 @@ def get_course_overview_with_access(user, action, course_key, check_if_enrolled=
     return course_overview
 
 
+def is_enrolled_in_course_or_staff(course_key, user):
+    """
+    Returns true if the user is enrolled in the specified course.
```

Comment:
I've decided to remove this method because it only adds confusion. It is easy enough to do the two checks if necessary.